src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class UnconditionalSuppressMessageAttributeState
+	{
+		private readonly Dictionary<IMetadataTokenProvider, Dictionary<string, SuppressMessageInfo>> _localSuppressionsByMdToken;
+
+		private bool HasLocalSuppressions {
+			get {
+				return _localSuppressionsByMdToken.Count != 0;
+			}
+		}
+
+		public UnconditionalSuppressMessageAttributeState ()
+		{
+			_localSuppressionsByMdToken = new Dictionary<IMetadataTokenProvider, Dictionary<string, SuppressMessageInfo>> ();
+		}
+
+		public void AddLocalSuppression (CustomAttribute ca, IMetadataTokenProvider mdTokenProvider)
+		{
+			SuppressMessageInfo info;
+			if (!TryDecodeSuppressMessageAttributeData (ca, out info)) {
+				return;
+			}
+
+			if (!_localSuppressionsByMdToken.TryGetValue (mdTokenProvider, out var suppressions)) {
+				suppressions = new Dictionary<string, SuppressMessageInfo> ();
+				_localSuppressionsByMdToken.Add (mdTokenProvider, suppressions);
+			}
+
+			suppressions[info.Id] = info;
```

Comment:
Please add a comment that we intentionally use the last one (this is potentially another candidate for a warning message - or maybe just informational - generating warning message from within suppression might cause trouble